src/bm_sim/P4Objects.cpp

```diff
@@ -2168,12 +2168,12 @@ P4Objects::init_objects(std::istream *is,
 
     init_errors(cfg_root);  // parser errors
 
+    init_calculations(cfg_root);
+
     init_parsers(cfg_root, &init_state);
 
     init_deparsers(cfg_root);
 
-    init_calculations(cfg_root);
-
```

Comment:
what's the reason for this change?

Reply:
Because in *init_parsers* or *init deparsers* we can have Chekcsum update primitive which needs a calculation object as parameter to compute checksum.